### ISSUE TITLE
Suggestion regarding long document with hierarchical bullet points  in German

### ISSUE THREAD

**Issue by sbhttchryy** [UKPLab/sentence-transformers#714]
Hello, a very big thank you for this wonderful library. I am a beginner, so sorry if my question is a bit naive. I have a collection of domain specific documents ( in German). The documents have a collection of hierarchical bullet points in them, e.g.: 

Cause:
xyz
abc
efg
    123
    456
hij

Which is why, I concatenated them to a long sentence. The sentences are long, some of them are ~5000 words long. I want to perform question answering in this document, which is why i was wondering if using a longformer be a a suitable case for my scenario.

My question are as follows:
1. How can i use a longformer in a cross-encoder / bi-encoder in my case?
2. I am reluctant to break up my sentences into shorter pieces as information might get lost, e.g. if I have [Cause: xyz abc efg 123] as a sentence, [456 hij] will lose information? Do you have any suggestion how to proceed in this case?

Thank you so much.

**Comment by nreimers** [comment]
Hi,
the usage of Longformer is quite hard, as you have to define how your attention mask looks like. BERT & co. perform attention between all tokens, but for Longformer you must set it by hand => this requires to know between which tokens you need attention.

Also I think dense approaches are the wrong approach to encode long text. BERT embeddigns to 768 dim, so this is 768*4 = 3072 bytes.

If you have a long text of maybe 50k characters, you must compress 50k bytes to 3072 bytes. Hence, the longer your text, the more information you will loose. And at some point the loss is too large. You cannot fit 50k/500k or 5M into just 3072 bytes.

I think splitting your data into smaller chunks of maybe 100-300 words make sense.

**Comment by happygirlzt** [comment]
Hi, 
I happened to want to ask this question as well. Regarding handling long text, I found two related pretrained models, i.e., Longformer and Bigbird. It seems that Longformer has already been included in Huggingface Transformer library (https://huggingface.co/transformers/model_doc/longformer.html). But I saw your last reply that using Longformer is hard. So I may want to try the other approach, which is splitting the data into smaller chunks. I am wondering do you have any example to split the long text (e.g., document) into shorter texts (e.g., sentences) to fine-tune with SBERT. Splitting the data is simple, however, I'm not sure how to fine-tune them for text similarity detection. Many thanks in advance!

**Comment by nreimers** [comment]
Hi @happygirlzt 
>  however, I'm not sure how to fine-tune them for text similarity detection. 

Sadly labeled data is required for this: You must have some positive pairs where you say these have a high similarity.

Unsupervised approaches sadly do not work that well.

**Comment by happygirlzt** [comment]
Hi @nreimers 
Thank you for your reply. I tried Longformer in the Transformers library and I found the usage is the same as other models (e.g, BERT, RoBERTa). Thus, I'm not sure why it is difficult to include Longformer in this SBERT framework. Thanks again!

**Comment by nreimers** [comment]
Hi @happygirlzt 
It was quite a long time ago I looked into longformer. If it is usable with AutoModel, it can be used without any changes in SBERT.

**Comment by happygirlzt** [comment]
Hi @nreimers 
Yes, I tried and Longformer is usable with AutoModel and AutoTokenizer. Thus, I wonder whether we can use Longformer in SBERT like `model = SentenceTransformer('distilbert-base-nli-stsb-mean-tokens')`, where 'distilbert-base-nli-stsb-mean-tokens' is a model pre-trained by you guys. However, if I am not wrong, there is no pre-trained Longformer model available in SBERT now. So, what is the possible way to use Longformer with SBERT? Many thanks again!

**Comment by hanshupe** [comment]
I wonder if there are any news regarding using Longformer with SBERT? Longformer is available on Huggingface and behaves the same way as the other models, but to use it for SentenceTransformer i guess it must be trained to get the SBERT sentence embeddings.
